include/xtensor/xview.hpp

```diff
@@ -609,9 +611,13 @@ namespace xt
     template <class CT, class... S>
     template <class T>
     inline auto xview<CT, S...>::strides() const ->
-        std::enable_if_t<has_raw_data_interface<T>::value, strides_type>
+        std::enable_if_t<has_raw_data_interface<T>::value, strides_type&>
     {
-        strides_type strides = xtl::make_sequence<strides_type>(m_e.dimension() - integral_count<S...>(), 0);
+        if (m_strides_computed)
+        {
+            return m_strides;
+        }
+        m_strides = xtl::make_sequence<strides_type>(dimension(), 0);
```

Comment:
This shows another bug in the computaion of the strides: the `new_axis` were not taken into account in the previous code. Now that you fixed that, you also need to fix the loop.